### DISCUSSION TITLE
Disable ktlint rules with ktlint-gradle version 12.0.2

### DISCUSSION THREAD

**Discussion by JohannesTeichert** [JLLeitschuh/ktlint-gradle#728] [Q&A] [answered]
I hope it is appropriate to ask for support here.

I upgraded the ktlint-gradle plugin to version 12.0.2 and now it does not respect the settings in my modules' build.gradle where I previously disabled some rules like that:

```
ktlint {
    ktlint.setDisabledRules(["package-name, max_line_length"])
}
```

I also tried using a .editorconfig file but that does not allow all the options I need. Disabling max_line_lenth did work there but I dont have the option for package-name.

![image](https://github.com/JLLeitschuh/ktlint-gradle/assets/15386957/c6b21757-3ab5-4379-883a-8b898284f305)

In the projects build.gradle I have the ktlint-gradle dependency in the buildscript and for allprojects I apply the ktlint-gradle plugin.

```
buildscript {
    ...
    dependencies {
        classpath 'com.android.tools.build:gradle:8.1.4'
        classpath "org.jlleitschuh.gradle:ktlint-gradle:12.0.2" // https://github.com/JLLeitschuh/ktlint-gradle/releases  
        ...
  }
}

allprojects {
    ...
    apply plugin: "org.jlleitschuh.gradle.ktlint"
}
```

I am not sure if this is a ktlint configuration problem, oder gradle, or android studio, or most likely me not understanding editorconfig correctly.

**Comment by wakingrufus** [comment, marked as answer]
I think you want:
ktlint_standard_package-name = disabled

See:
https://pinterest.github.io/ktlint/latest/faq/#why-is-editorconfig-property-disabled_rules-deprecated-and-how-do-i-resolve-this

https://pinterest.github.io/ktlint/dev-snapshot/rules/standard/

**Reply by JohannesTeichert** [reply]
oh wow yes thank you so much, this worked, the IDE did not recognise these as supported properties. But while running ktlintcheck its taken into account.
